stingray/crossspectrum.py

```diff
@@ -778,20 +778,24 @@ def _make_segment_spectrum(self, lc1, lc2, segment_size):
 
         # In case a small difference exists, ignore it
         lc1.dt = lc2.dt
-        if self.gti is None:
-            self.gti = cross_two_gtis(lc1.gti, lc2.gti)
-            lc1.gti = lc2.gti = self.gti
-            lc1._apply_gtis()
-            lc2._apply_gtis()
+
+        gti = cross_two_gtis(lc1.gti, lc2.gti)
+        lc1._apply_gtis()
+        lc2._apply_gtis()
+        if self.gti is None or np.all(self.gti == gti):
```

Comment:
I don't understand the logic here. If they are already identical, why do we assign `gti` to `self.gti`?

Reply:
Because it made sense in my head? :) 

Mostly I wanted the same behaviour for the case when `self.gti == None` and when the input GTI is the same as the one already assigned to `self.gti`. 

I've appended the logic to hopefully be more sensible.